src/components/CompareResults/CommonGraph.tsx

```diff
@@ -0,0 +1,131 @@
+import { Chart as ChartJS, LineElement, LinearScale } from 'chart.js';
+import 'chart.js/auto';
+import * as kde from 'fast-kde';
+import { Line } from 'react-chartjs-2';
+import { style } from 'typestyle';
+
+import { Spacing } from '../../styles';
+
+ChartJS.register(LinearScale, LineElement);
+
+const styles = {
+  container: style({
+    display: 'flex',
+    marginBottom: Spacing.Medium,
+    width: '100%',
+  }),
+};
+
+function CommonGraph(props: CommonGraphProps) {
+  const { baseRevisionRuns, newRevisionRuns } = props;
+
+  const options = {
+    plugins: {
+      legend: {
+        align: 'start' as const,
+        position: 'bottom' as const,
+      },
+      title: {
+        align: 'start' as const,
+        display: true,
+        text: 'Runs Density Distribution',
+      },
+    },
+    responsive: true,
+    scales: {
+      x: {
+        grid: {
+          display: false,
+          offset: false,
+        },
+        type: 'linear' as const,
+        title: {
+          align: 'end' as const,
+          display: true,
+          text: 'Run Result' as const,
+        },
+      },
+      y: {
+        beginAtZero: true,
+        grid: {
+          drawBorder: false,
+          display: false,
+          offset: false,
+        },
+        ticks: {
+          beginAtZero: true,
+          display: true,
+        },
+        title: {
+          align: 'end' as const,
+          display: true,
+          text: 'Run Density' as const,
+        },
+      },
+    },
+  };
+
+  //////////////////// START FAST KDE ////////////////////////
+  const baseRunsDensity = Array.from(
+    kde.density1d([...baseRevisionRuns.values], {
```

Comment:
Are you sure you need to spread this array here? I think you could pass the values directly.

Spreading the array creates a new array (clones it), which is desirable sometimes, to avoid it being modified for example, or when it's not a real array initially (Map / Set / DOMList..., because it uses the iterator protocol underneath). In this case I believe it's not necessary. This is not a big deal with the array is small, but could bring slowdowns otherwise (either from the copy itself, or from the GC collecting the copy after the function runs).

The same applies below for `newRunsDensity`.

To be explicit, the spread in `Math.min` and `Math.max` is completely different: it's not spreading to create a _new_ array, it's spreading to argument functions. That one has other issues: if there are too many values (to be honest, I don't remember the limit, but it's very high), the javascript engine will error. I think we're safe here in this case.